### PR TITLE
chore(dashboard): update react scripts to version 3.1.2

### PR DESCRIPTION
**Pull request by eysi09**
**What this PR does / why we need it**:

Updates the Dashboard react-scripts deps. 

**Special notes for your reviewer**:

It was necessary to re-configure the lint setup and to address some lint
errors that came up, most notably the react-hooks/exhaustive-deps. To
address this we added the useMountEffect helper function.

### CODE REVIEW

File: tslint.base.json [hunk 1]
```diff
@@ -6,6 +6,11 @@
   "rulesDirectory": [
     "node_modules/tslint-microsoft-contrib"
   ],
+  "linterOptions": {
+    "exclude": [
+      "./dashboard/**/*"
+    ]
+  },
```
Comment (edvald) [inline tslint.base.json]: I don't think this is necessary, since we don't lint from the root project anymore.

Comment (eysi09) [reply tslint.base.json]: I couldn't get linting to work in VSCode until I removed this. I'll went back on forth on some configuration options though so I maybe something else was configured differently when it wasn't working. I'll verify and report back.

Comment (eysi09) [reply tslint.base.json]: So yeah, if we don't exclude the dashboard, we get lint warnings in VSCode from `tslint.base.json`. But the dashboard doesn't use those lint rules, in fact in can't. Instead we define our own lint rules for the dashboard. I've made some changes to those rules in #1234.

Comment (eysi09) [reply tslint.base.json]: Actually, I've found away around this in #1234. But it requires some other changes so I'll leave this as is and address it there.

**Pull request merged**